cmd/fix-usernames/main.go

```diff
@@ -0,0 +1,319 @@
+	}
+
+	// Load the most recent data usage record for the user.
+	usage, err := loadMostRecentDataUsage(ctx, tx, oldUser.Username, newUser.Username)
+	if err != nil {
+		return errors.Wrapf(
+			err,
+			"unable to load the most recent data usage for %s and %s",
+			oldUser.Username,
+			newUser.Username,
+		)
+	}
+
+	// Deactivate all plans for both the old username and the new username.
+	err = db.DeactivateUserPlans(ctx, tx, *oldUser.ID)
+	if err != nil {
+		return errors.Wrapf(err, "unable to deactivate existing plans for %s", oldUser.Username)
+	}
+	err = db.DeactivateUserPlans(ctx, tx, *newUser.ID)
+	if err != nil {
+		return errors.Wrapf(err, "unable to deactivate existing plans for %s", newUser.Username)
+	}
+
+	// Subscribe the new user to the default plan if they didn't have a current subscription.
+	var newSubscription *model.UserPlan
+	if oldSubscription == nil {
+		newSubscription, err = db.SubscribeUserToDefaultPlan(ctx, tx, newUser.Username)
+		if err != nil {
+			return errors.Wrapf(err, "unable to subscribe %s to the default plan", newUser.Username)
+		}
+	} else {
+		plan := &oldSubscription.Plan
+		newSubscription, err = db.SubscribeUserToPlan(ctx, tx, newUser, plan)
+		if err != nil {
+			return errors.Wrapf(err, "unable to subscribe %s to the %s plan", newUser.Username, plan.Name)
+		}
+	}
+
+	// Get all of the details for the new subscription.
+	newSubscription, err = loadSubscription(ctx, tx, *newSubscription.ID)
+	if err != nil {
+		return errors.Wrapf(err, "unable to load the new subscription details for %s", newUser.Username)
+	}
+
+	// Ensure that the new quotas are greater than or equal to the old quotas if applicable.
+	if oldSubscription != nil {
+		err = restorePreviousQuotas(ctx, tx, oldSubscription, newSubscription)
+		if err != nil {
+			return errors.Wrapf(err, "unable to restore previous quotas for %s", newUser.Username)
+		}
+	}
+
+	// Associate the usage with the current subscription.
+	if usage != nil {
+		err = addUsageToSubscription(ctx, tx, newSubscription, usage)
+		if err != nil {
+			return errors.Wrapf(err, "unable to add data usage to the new subscription for %s", newUser.Username)
+		}
+	}
+
+	return nil
+}
+
+func main() {
+
+	// Load the configuration.
+	cfg, err := loadConfig()
+	if err != nil {
+		log.Fatalf("unable to load the configuration: %s", err)
+	}
+
+	// Establish the database connection.
+	_, gormdb, err := db.Init("postgres", cfg.DatabaseURI)
+	if err != nil {
+		log.Fatalf("unable to connect to the database: %s", err)
+	}
+
+	// Run the actual updates in a transaction.
+	err = gormdb.Transaction(func(tx *gorm.DB) error {
+		ctx := context.Background()
+
+		// Get the list of usernames with suffixes.
+		usersToFix, err := listUsersWithSuffixes(ctx, tx, cfg.UsernameSuffix)
+		if err != nil {
+			return errors.Wrap(err, "unable to list usernames with suffixes")
+		}
+
+		// Just list the usernames for now.
```

Comment:
Since `fixUsername` is getting called below, it looks like this comment is now outdated?

Reply:
Ah, good catch! I'll fix it.